### ISSUE TITLE
Feature request: load an array of background page scripts like chrome extensions do 

### ISSUE THREAD

**Issue by SchizoDuckie** [nwjs/nw.js#5769]
With the post-0.13 improvements and nwjs moving ever closer to Chrome i've decided to adopt adopt more of chrome's background page features using the manifest.json

![image](https://cloud.githubusercontent.com/assets/111710/23835384/ae531c18-0766-11e7-8e90-f46c8b6fcbde.png)

This all works beautifully, and I can achieve the same result with nwjs by concatting all these scripts into one file and loading it.

However, this introduces a new build step where you have to do the concatting manually before the first run.

To make it dynamic, I tried hacking this up:

![image](https://cloud.githubusercontent.com/assets/111710/23835404/f28bac38-0766-11e7-8345-185a2676f162.png)

This works, *if* your system is fast enough. Otherwise async load timeout dependencies make it fail.

I also tried to cheat by moving this script to `node-main` in `package.json`, to have it dynamically write my background bootstrap page to file and load that in the regular `bg-script`. 
This fails because by the time `node-main` runs the `bg-script` is also already loaded
(therefore it will only have an effect on the next run, which is another PITA)

Long story short:

## Proposal
A new `array[]` item for package.json: `bg-scripts` that loads these files in sequence and basically does exactly what both chrome itself and my example script and  do:

![image](https://cloud.githubusercontent.com/assets/111710/23835484/5b0a8cce-0768-11e7-9f60-f3908b4daefe.png)

which results in this generated background page:
![image](https://cloud.githubusercontent.com/assets/111710/23835468/2bb904c8-0768-11e7-9e39-de4a9d477e46.png)

**Comment by pdrd** [comment]
Why are your dependencies failing with your dynamic approach? Isn't it an option to simply use a module loader like system.js? You can also run it in a 'hybrid' mode, where you can load scripts via <script>-tags and node-modules and you shouldn't have any dependency failures..

Ever thought about using [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) for your background tasks? They are *independent* to chrome and therefore more portable if you want to reuse your code.

**Comment by SchizoDuckie** [comment]
I'm not using any module loaders at the moment. the project is checkout-and-run, scripts are minified only when building a nightly or production version. I have no current plans to switch to system.js or webpack, or introduce a mandatory CLI build step.

The dependencies seem to be failing because my trick to parse `manifest.json`'s background script entry fails to load all scripts before giving the 'background page loaded' event, thereby causing random timeouts in nwjs context on some runs when the foreground page is loaded (not all though) 

I'm not sure what you mean by using '<script>' tags. The bg-scripts requires a .js file afaik, not an html file.

Web workers will not prevent any database migrations I may have from finishing when a user closes the tab they've just opened, causing the db to be in an invalid state (my codebase runs as a chrome extension in 2 different forms as well as standalone using nwjs)

By moving to the SQL execution to the background page and only initializing my WebSQL database and and any SQL db migrations that may need to run there, this process is automagically handled when the extension auto-updates, and the user does not have to wait for it to complete before navigating away from a freshly opened tab. being able to use the same system for both nwjs and chrome is a way to reduce extra  complexity in an already complex web-app.

**Comment by pdrd** [comment]
Alright, got it! Thats true, the web workers won't fit your needs, but you might be able to expose the functions to the browser context by using 'exports' in your nw.js-background-scripts, which you can call in the browser context. Even if your tab is closed, the default background page is still active and can finish your sql tasks. But this is not fulfilling your wish to be able to use your code in both systems.

Well, you should consider using a dynamic module loader like system.js. It does *not* require an additional build step like webpack, it will still be 'checkout-and-run', because if run in a node context it calls internally the node 'require' and in browser context it adds a <script>-tag, but you simply don't have to worry about dependency resolving issues.

Have you tried to implement an event that is triggered when the background page is ready?

**Comment by SchizoDuckie** [comment]
Sorry, i'm not going into dynamic script loading. That's way out of scope for this simple issue.

An event sent from the background page to the foreground doesn't make much sense either, that's exactly what the background page is supposed to do.

For now we have mitigated the problem by adding the build step that concats the scripts from manifest.json. 

I cannot use exports because the communcations from the background page go through the [chrome events channels](https://github.com/SchizoDuckie/DuckieTV/blob/angular/js/vendor/CRUD.BackgroundPageAdapter.js) dedicated to this exact feature (communicating from foreground to background and vice versa) 

However, this does require all components of my WebSQL ORM to be loaded and initialized so the database can be opened. There is no need for any other dynamic loading if the platform does it for you, and if nwjs moves one step closer to chrome (or can even piggyback on the same code that chrome itself uses) that's a win for us both.
